test/test_nn.py

```diff
@@ -6144,6 +6144,72 @@ def test_transformerencoderlayer(self):
         self.assertEqual(tuple(result.shape), tuple(ref_output.shape))
         np.testing.assert_allclose(result, ref_output, atol=1e-5)
 
+    def test_transformerencoderlayer_gelu(self):
+        # this is a deterministic test for TransformerEncoderLayer with gelu activation
+        d_model = 4
+        nhead = 2
+        dim_feedforward = 16
+        dropout = 0.0
+        bsz = 2
+        activation = "gelu"
+
+        model = nn.TransformerEncoderLayer(d_model, nhead, dim_feedforward, dropout, activation)
+
+        # set constant weights of the model
+        for idx, p in enumerate(model.parameters()):
+            x = p.data
+            sz = x.view(-1).size(0)
+            shape = x.shape
+            x = torch.cos(torch.arange(0, sz).float().view(shape))
+            p.data.copy_(x)
+
+        # deterministic input
+        encoder_input = torch.Tensor([[[20, 30, 40, 50]]])
+        result = model(encoder_input)
+        ref_output = torch.Tensor([[[2.249815, 0.131006, -0.702199, 0.177868]]])
+        result = result.detach().numpy()
+        ref_output = ref_output.detach().numpy()
+        self.assertEqual(tuple(result.shape), tuple(ref_output.shape))
+        np.testing.assert_allclose(result, ref_output, atol=1e-5)
+
+        # deterministic input
+        encoder_input = torch.Tensor([[[1, 2, 3, 4]],
+                                      [[5, 6, 7, 8]]])
+        result = model(encoder_input)
+        ref_output = torch.Tensor([[[2.264103, 0.121417, -0.696012, 0.159724]],
+                                   [[2.264103, 0.121417, -0.696012, 0.159724]]])
+        result = result.detach().numpy()
+        ref_output = ref_output.detach().numpy()
+        self.assertEqual(tuple(result.shape), tuple(ref_output.shape))
+        np.testing.assert_allclose(result, ref_output, atol=1e-5)
```

Comment:
If you have to use numpy, please decorate the test to be skipped if numpy hasn't been installed. Also there is an allclose method for pytorch as well. We prefer if you use that. See other tests for examples.

Reply:
I use torch.testing.assert_allclose in the tests